### PR TITLE
Section on Go Makefile standards

### PR DESCRIPTION
**Pull request by GeorgeMac**
This section will document how we structure our Makefiles. It may also document a framework or tool we use to initialize a Makefile for a new Go project.

### CODE REVIEW

File: makefiles/examples/Makefile [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+test: ## Run all the tests
+	echo 'mode: atomic' > coverage.txt && go list ./... | grep -v /vendor/ | xargs -n1 -I{} sh -c 'go test -covermode=atomic -coverprofile=coverage.txt -v -race -timeout=30s {}'
+
+.PHONY: cover
+cover: test ## Run all the tests and opens the coverage report
+	go tool cover -html=coverage.txt
+
+.PHONY: fmt
+fmt: ## gofmt and goimports all go files
+	find . -name '*.go' -not -wholename './vendor/*' | while read -r file; do gofmt -w -s "$$file"; goimports -w "$$file"; done
```
Comment (GeorgeMac) [inline makefiles/examples/Makefile]: Is this now solved thanks to the updates which mean `./...` now exclude the vendor dir? Or do we need to upgrade to go1.9 first?

**Comment by GeorgeMac** [comment]
p.s. merge in master for spellchecking

File: makefiles/examples/Makefile (hunk 1 above)
Comment (markphelps) [reply makefiles/examples/Makefile]: will test

Comment (markphelps) [reply makefiles/examples/Makefile]: still doesnt work :(

Althought since `goimports` also `gofmts`, we can remove the call to `gofmt`

**Comment by GeorgeMac** [comment]
I was going to approve it, but then realized I opened it and can't but 👍 all the same

**Pull request merged**